### ISSUE TITLE
@Mentioning view not correct

### ISSUE THREAD

**Issue by steefzz** [mantisbt-plugins/BBCodePlus#62]
When I want to notify someone with @ (@username), I
 see the (not clickable) uri of the user:
https://bugtracker.host.nl/view_user_page.php?id=44

instead of  @username

MantisBT 2.21.0
BBCode+ 2.1.3
PHP version 7.1.18

![2019-04-23 19_26_26-Window](https://user-images.githubusercontent.com/13647902/56602616-2b0ece80-65fe-11e9-81d7-462bb3f200fe.png)

**Comment by tiele004** [comment]
Bumping into this issue because I have te same issue here. 
Looking for a solution quite a while now so help would be very much appreciated.

**Comment by bkraul** [comment]
A new set of changes has been merged into the master. Please download master and check for correction of this issue with feedback.

**Comment by bkraul** [comment]
Found a bug with the mentioning link related to the regex. Working to resolve.

**Comment by bkraul** [comment]
Corrected the issue. Will await for a few days for further feedback before closing the issue.

**Comment by steefzz** [comment]
@bkraul 
This issue seems to be fixed with version 2.1.6
Thanks

**Comment by AmauryBs** [comment]
I still have the problem with bug links of resolved bugs (bug_resolved_status_threshold <= bug status ) it shows /support/view.php?id=5000 in plain text instead of the link. Mantis 2.21.0 + BBcode plus vers 2.1.6

**Comment by bkraul** [comment]
I am not sure I am understanding. Could you please post a screenshot of what your referenced bug looks like? I have marked a bug as resolved, and then referenced it with #[ID] in another bug, and the link showed correctly.

Do you have **MantisCoreFormatting** plugin disabled? Latest changes to the BBCodePlus plugin no longer try to generate bug links. Which means that if you want that functionality, you must enable the MantisCoreFormatting plugin, which is the one that provides it.

**Comment by AmauryBs** [comment]
I can't send a screen right now but I will send it tomorow. I do have MantisCoreFormatting plugin enabled since comment link ~[comment_id] and buglink on non-resolved Mantis works. Bug links for resolved bugs are displayed differently (they are striked) and the bug might come from that.
 I temporarily solved this by displaying resolved bugs link like other bugs link (i.e not striked) in core/string_api.php. I will also send a screenshot of this tomorow.

**Comment by bkraul** [comment]
Yeah, please do. That way I will be able to replicate and fix if possible.

**Comment by bkraul** [comment]
@AmauryBs I was able to replicate your issue. There is no need to send me a screenshot. I will look at the problem and attempt to resolve.

![image](https://user-images.githubusercontent.com/14616851/61411828-e5843980-a8ac-11e9-9f98-e34869a1c4c7.png)

**Comment by bkraul** [comment]
@AmauryBs please confirm correction of behavior by downloading the latest master or release (2.1.7).

**Comment by AmauryBs** [comment]
@bkraul the fix worked in some way but broked all the bbcode tags and the display of text in the whole site ( the screenshots bellow show Mantis before and after the fix) 
![before_plugin](https://user-images.githubusercontent.com/43411870/61439990-d780f800-a942-11e9-8602-7d73c26aa681.png)
![bug_bbcode_plugin](https://user-images.githubusercontent.com/43411870/61439991-d8198e80-a942-11e9-92df-744427a3432e.PNG)
![bug_link](https://user-images.githubusercontent.com/43411870/61439993-d8198e80-a942-11e9-92c9-af7d127ccae1.png)
![expected view](https://user-images.githubusercontent.com/43411870/61439994-d8198e80-a942-11e9-876f-5ed305b61ed8.png)
![my view](https://user-images.githubusercontent.com/43411870/61439995-d8198e80-a942-11e9-89a0-f3c2c8d2a2b8.png)

**Comment by AmauryBs** [comment]
ok i found what it was, you forgot to remove a echo at line 297 in BBCodePlus.php. It seems to work fine without it, I will test it a bit more and tell you if I find anything else. Thanks for this quick fix btw

**Comment by bkraul** [comment]
Ah crap. You are right. Will remove that asap.

**Comment by bkraul** [comment]
It is done. v2.1.8 is available.

**Comment by merlin-tc** [comment]
Mentioning people with "@" and links to other tickets via # and ~ are working with v2.1.8.
What's still not working are hyperlinks. When I disable the plugin the hyperlinks work.
I attached screenshots of all relevant options. If something is missing let me know.

![bbcode](https://user-images.githubusercontent.com/2970561/61547347-7db82500-aa4b-11e9-8004-8288e9d091d7.png)

![format](https://user-images.githubusercontent.com/2970561/61547356-827cd900-aa4b-11e9-8f24-1e317082c275.png)

![ticket](https://user-images.githubusercontent.com/2970561/61547363-86a8f680-aa4b-11e9-8f7c-aaa335344d8d.png)

**Comment by bkraul** [comment]
That is due to the an interaction of automatic link creation and the BBCode implementation. This is an unavoidable behavior. Take for instance:

you have `[img]https://proxy.duckduckgo.com/iu/?u=http%3A%2F%2Fs7d2.scene7.com%2Fis%2Fimage%2FPetSmart%2FARTHMB-HowToHelpYourCatHaveFewerHairballs-20160818%3F%24AR1104%24&f=1[/img]`, which is supposed to generate an image like:

![image](https://user-images.githubusercontent.com/14616851/61548472-665d4580-aa13-11e9-833b-2d3f77d699c2.png)

If automatic link generation is applied, what you will end up with the following markup:
```html
<img src="<a target=" _blank"="" href="https://proxy.duckduckgo.com/iu/?u=http%3A%2F%2Fs7d2.scene7.com%2Fis%2Fimage%2FPetSmart%2FARTHMB-HowToHelpYourCatHaveFewerHairballs-20160818%3F%24AR1104%24&amp;f=1[/img">]https://proxy.duckduckgo.com/iu/?u=http%3A%2F%2Fs7d2.scene7.com%2Fis%2Fimage%2FPetSmart%2FARTHMB-HowToHelpYourCatHaveFewerHairballs-20160818%3F%24AR1104%24&amp;f=1"&gt;
```
Which would obviously break image display.

[This](https://github.com/mantisbt-plugins/BBCodePlus/blob/1c581aee6dcc9e53e4b9e76f7b4e8b9c49797990/BBCodePlus/BBCodePlus.php#L299) is the line that prevents it:

Therefore, in order to declare loose links, you need to surround them with `[url]http://yourlink.com[/url]` tags.

**Comment by merlin-tc** [comment]
Is there any way to workaround this?
I have some users for whom it is in impossible task to add the url tags :-(

For example if URL processing in the mantis formating plugin is disabled, could it be added automatically by the bbcode plugin?

**Comment by bkraul** [comment]
I might be able to do some regex sorcery, but it's gonna take me a while. I am not an expert in regex 😞 ...

**Comment by merlin-tc** [comment]
@FSD-Christian-ISS just sent you a pull requests, we are using this version internally now.

**Comment by bkraul** [comment]
Fixed via #78.
